### PR TITLE
Tendermint update

### PR DESCRIPTION
**Pull request by zhangyelong**
## Ethermint can not be connected to Tendermint v0.15.0, because ABCI interface changed. 
Relates to: 
*[#377](https://github.com/tendermint/ethermint/issues/377)*

## Checklist
- [ ] Update versions in glide.yaml, glide.lock
- [ ] Update abci implementions in app/, cmd/, ethereum/

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -232,53 +262,64 @@ func (app *EthermintApplication) validateTx(tx *ethTypes.Transaction) abciTypes.
 	// Make sure the transaction is signed properly
 	from, err := ethTypes.Sender(signer, tx)
 	if err != nil {
-		return abciTypes.ErrBaseInvalidSignature.
-			AppendLog(core.ErrInvalidSender.Error())
+		// TODO: Add errors.CodeTypeInvalidSignature ?
```
Comment (melekes) [inline app/app.go:265]: mm.. I think ethermint should have its own set of error codes

similar to cosmos https://github.com/cosmos/cosmos-sdk/blob/master/errors/code.go#L4, but yet specific to ethermint application

**Comment by melekes** [comment]
awesome work @zhangyelong! just one note related to using cosmos errors

File: app/app.go (hunk 1 above)
Comment (adrianbrink) [reply app/app.go:265]: Yes, but I think we should update those in a separate PR.

**Comment by zhangyelong** [comment]
Thanks for replying, what else can I do for this PR ?

File: app/app.go [hunk 2]
```diff
@@ -196,32 +225,33 @@ func (app *EthermintApplication) Query(query abciTypes.RequestQuery) abciTypes.R
 	app.logger.Debug("Query") // nolint: errcheck
 	var in jsonRequest
 	if err := json.Unmarshal(query.Data, &in); err != nil {
-		return abciTypes.ResponseQuery{Code: abciTypes.ErrEncodingError.Code,
+		return abciTypes.ResponseQuery{Code: errors.CodeTypeInternalErr,
```
Comment (ebuchman) [inline app/app.go:228]: should be encoding error

**Review by ebuchman — APPROVED** [review]

**Pull request merged**